text/0000-arguments.md

```diff
@@ -120,16 +125,18 @@ The `launcher` currently has a three step command resolution flow:
 
 1.  0 args: execute the default process type, where the default process type is `${CNB_PROCESS_TYPE:-web}`
 1.  1 arg:  if `$1` matches a process type, execute that process type
-1.  N args: execute `$1` as the command with `${@:2}` as the arguments appended to that command
+1.  N args if `$1` is `--`: directly execute `$2` as the command with `${@:3}` as the arguments appended to that command
+1.  N args: execute `$1` as the command with `${@:2}` as the arguments appended to that command in a `bash` shell
 
-This proposal is to augment that command resolution flow to:
+This proposal change the command resolution flow to:
 
 1.  If `CNB_PROCESS_TYPE=override`
     1. 0 args: fail
-    1. N args: execute `$1` as the command with `${@:2}` as the arguments
+    1.  N args if `$1` is `--`: directly execute `$2` as the command with `${@:3}` as the arguments appended to that command
+    1.  N args: execute `$1` as the command with `${@:2}` as the arguments appended to that command in a `bash` shell
 1.  Else
     1. 0 args: execute the process type `${CNB_PROCESS_TYPE:-web}`
-    1. N args: Execute process type `${CNB_PROCESS_TYPE:-web}`, after appending `"$@"` to the arguments
+    1. N args: Execute process type `${CNB_PROCESS_TYPE:-web}`, after appending `$@` (all positional arguments) to the process arguments
```

Comment:
Should the `"` be removed from this?  Ignoring, for a moment, the actual implementation, is your point that with this system, property tokenization is already handled and it's no longer a concern that each argument is properly quoted for tokenization?

Reply:
I just wanted a shorthand for all args. Adding the quotes seems to imply something about the implementation. Maybe we should just skip the bash shorthand in this section all together? It potentially get's confusing in context.

Reply:
I am trying to describe appending to an array of strings here and with the quotes it seems to imply something else? But I'll defer to my readers

Reply:
Having the quotes is a specific bash-ism that implies that the shell will properly quote everything so a later tokenization gets the same collection.  Without it, arguments with spaces would be re-tokenized as multiple arguments.  So the reason for me originally adding that syntax is to indicate that we would do the appropriate things necessary to ensure that the original tokenization would be preserved through out.  It may not be necessary any longer, if you believe the proposal already accommodates that behavior before describing it here.